dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeVectorStoreRecordCollection.cs

```diff
@@ -347,13 +403,44 @@ private async Task RunOperationAsync(string operationName, Func<Task> operation)
         }
     }
 
-    private async Task<Sdk.Index<GrpcTransport>> GetIndexAsync(string indexName, CancellationToken cancellationToken)
+    private IndexClient GetIndexClient()
     {
-        this._index ??= await this._pineconeClient.GetIndex(indexName, cancellationToken).ConfigureAwait(false);
+        if (this._indexClient is null)
+        {
+#if NON_PUBLISH
```

Comment:
Rather than a different build, shouldn't we just allow uesrs to set the host on PineconeVectorStoreOptions, and use that here? It'd be null by default (so good for production), but some users may want to actually use the emulator themselves with the connector, and would set PineconeVectorStoreOptions.Host for that?